daemon/images.go

```diff
@@ -41,28 +42,35 @@ func (d *Daemon) pollForNewImages(logger log.Logger) {
 	changes := &update.Automated{}
 	for _, service := range services {
 		for _, container := range service.ContainersOrNil() {
-			logger := log.With(logger, "service", service.ID, "container", container.Name, "currentimage", container.Image)
-
 			currentImageID := container.Image
-			if err != nil {
-				logger.Log("error", err)
-				continue
-			}
-
 			pattern := policy.GetTagPattern(candidateServicesPolicyMap, service.ID, container.Name)
 			repo := currentImageID.Name
-			logger.Log("repo", repo, "pattern", pattern)
+			logger := log.With(logger, "service", service.ID, "container", container.Name, "repo", repo, "pattern", pattern, "current", currentImageID)
 
 			filteredImages := imageRepos.GetRepoImages(repo).Filter(pattern)
 
 			if latest, ok := filteredImages.Latest(); ok && latest.ID != currentImageID {
 				if latest.ID.Tag == "" {
-					logger.Log("msg", "untagged image in available images", "action", "skip", "available", repo)
+					logger.Log("warning", "untagged image in available images", "action", "skip")
+					continue
+				}
+				if latest.CreatedAt.IsZero() {
```

Comment:
Because of the sort order, if there is any image with a non-zero `CreatedAt` then it will be considered newer than a zero-timestamp image. So if there is some weirdness going on where the `CreatedAt` timestamp of an image comes and goes, then this check does not prevent 'flipping'.

The ultra-cautious approach would be to skip updating if _any_ of the `filterImages` have a zero `CreatedAt`.

Reply:
Well, we do look through the whole back catalogue, so may as well check this.